<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<html><head><title>R: Epigenomic tools</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8">
<link rel="stylesheet" type="text/css" href="R.css">
</head><body>
<h1> Epigenomic tools
<img class="toplogo" src="../../../doc/html/logo.jpg" alt="[R logo]"></h1>
</h1>
<hr>
<div align="center">
<a href="../../../doc/html/packages.html"><img src="../../../doc/html/left.jpg" alt="[Up]" width="30" height="30" border="0"></a>
<a href="../../../doc/html/index.html"><img src="../../../doc/html/up.jpg" alt="[Top]" width="30" height="30" border="0"></a>
</div><h2>Documentation for package &lsquo;Repitools&rsquo; version 1.87</h2>

<ul><li><a href="../DESCRIPTION">DESCRIPTION file</a>.</li>
<li><a href="../doc/index.html">Overview of user guides and package vignettes</a>; browse <a href="../doc">directory</a>.</li>
</ul>

<h2>Help Pages</h2>


<p align="center">
<a href="#A">A</a>
<a href="#B">B</a>
<a href="#C">C</a>
<a href="#D">D</a>
<a href="#E">E</a>
<a href="#F">F</a>
<a href="#G">G</a>
<a href="#L">L</a>
<a href="#M">M</a>
<a href="#N">N</a>
<a href="#O">O</a>
<a href="#P">P</a>
<a href="#R">R</a>
<a href="#S">S</a>
<a href="#T">T</a>
<a href="#W">W</a>
<a href="#misc">misc</a>
</p>


<h2><a name="A">-- A --</a></h2>

<table width="100%">
<tr><td width="25%"><a href="annoDF2GR.html">annoDF2GR</a></td>
<td>Convert a 'data.frame' to a 'GRanges'.</td></tr>
<tr><td width="25%"><a href="annoDF2GR.html">annoDF2GR-method</a></td>
<td>Convert a 'data.frame' to a 'GRanges'.</td></tr>
<tr><td width="25%"><a href="annoGR2DF.html">annoGR2DF</a></td>
<td>Convert an annotated 'GRanges' to a 'data.frame'.</td></tr>
<tr><td width="25%"><a href="annoGR2DF.html">annoGR2DF-method</a></td>
<td>Convert an annotated 'GRanges' to a 'data.frame'.</td></tr>
<tr><td width="25%"><a href="annotationBlocksCounts.html">annotationBlocksCounts</a></td>
<td>Counts the number of sequencing reads within supplied genomic blocks.</td></tr>
<tr><td width="25%"><a href="annotationBlocksCounts.html">annotationBlocksCounts-method</a></td>
<td>Counts the number of sequencing reads within supplied genomic blocks.</td></tr>
<tr><td width="25%"><a href="annotationBlocksLookup.html">annotationBlocksLookup</a></td>
<td>Forms a mapping between probe locations and chromosomal blocks (regions).</td></tr>
<tr><td width="25%"><a href="annotationBlocksLookup.html">annotationBlocksLookup-method</a></td>
<td>Forms a mapping between probe locations and chromosomal blocks (regions).</td></tr>
<tr><td width="25%"><a href="annotationCounts.html">annotationCounts</a></td>
<td>Counts the number of sequencing reads surrounding supplied annotations</td></tr>
<tr><td width="25%"><a href="annotationCounts.html">annotationCounts-method</a></td>
<td>Counts the number of sequencing reads surrounding supplied annotations</td></tr>
<tr><td width="25%"><a href="annotationLookup.html">annotationLookup</a></td>
<td>Forms a mapping between probes on a tiling array and windows surrounding the TSSs of genes.</td></tr>
<tr><td width="25%"><a href="annotationLookup.html">annotationLookup-method</a></td>
<td>Forms a mapping between probes on a tiling array and windows surrounding the TSSs of genes.</td></tr>
</table>

<h2><a name="B">-- B --</a></h2>

<table width="100%">
<tr><td width="25%"><a href="BAM2GRanges.html">BAM2GRanges</a></td>
<td>Read in a (list of) BAM file(s) into a GRanges(List) object.</td></tr>
<tr><td width="25%"><a href="BAM2GRanges.html">BAM2GRanges-method</a></td>
<td>Read in a (list of) BAM file(s) into a GRanges(List) object.</td></tr>
<tr><td width="25%"><a href="BAM2GRanges.html">BAM2GRangesList</a></td>
<td>Read in a (list of) BAM file(s) into a GRanges(List) object.</td></tr>
<tr><td width="25%"><a href="BAM2GRanges.html">BAM2GRangesList-method</a></td>
<td>Read in a (list of) BAM file(s) into a GRanges(List) object.</td></tr>
<tr><td width="25%"><a href="FastQC-class.html">Basic_Statistics</a></td>
<td>FastQC and associated classes</td></tr>
<tr><td width="25%"><a href="FastQC-class.html">Basic_Statistics-method</a></td>
<td>FastQC and associated classes</td></tr>
<tr><td width="25%"><a href="binPlots.html">binPlots</a></td>
<td>Create line plots of averaged signal across a promoter</td></tr>
<tr><td width="25%"><a href="binPlots.html">binPlots-method</a></td>
<td>Create line plots of averaged signal across a promoter</td></tr>
<tr><td width="25%"><a href="ChromaResults-class.html">blocks</a></td>
<td>ChromaResults class</td></tr>
<tr><td width="25%"><a href="ChromaResults-class.html">blocks-method</a></td>
<td>ChromaResults class</td></tr>
<tr><td width="25%"><a href="blocksStats.html">blocksStats</a></td>
<td>Calculate statistics for regions in the genome</td></tr>
<tr><td width="25%"><a href="blocksStats.html">blocksStats-method</a></td>
<td>Calculate statistics for regions in the genome</td></tr>
</table>

<h2><a name="C">-- C --</a></h2>

<table width="100%">
<tr><td width="25%"><a href="checkProbes.html">checkProbes</a></td>
<td>Check Probe Specificity for Some Regions</td></tr>
<tr><td width="25%"><a href="checkProbes.html">checkProbes-method</a></td>
<td>Check Probe Specificity for Some Regions</td></tr>
<tr><td width="25%"><a href="chr21genes.html">chr21genes</a></td>
<td>Positions of Genes on Human Chromosome 21</td></tr>
<tr><td width="25%"><a href="ChromaBlocks.html">ChromaBlocks</a></td>
<td>A function to find areas of enrichment in sequencing data</td></tr>
<tr><td width="25%"><a href="ChromaBlocks.html">ChromaBlocks-method</a></td>
<td>A function to find areas of enrichment in sequencing data</td></tr>
<tr><td width="25%"><a href="ChromaResults-class.html">ChromaResults</a></td>
<td>ChromaResults class</td></tr>
<tr><td width="25%"><a href="ChromaResults-class.html">ChromaResults-class</a></td>
<td>ChromaResults class</td></tr>
<tr><td width="25%"><a href="ChromaResults-class.html">class:ChromaResults</a></td>
<td>ChromaResults class</td></tr>
<tr><td width="25%"><a href="FastQC-class.html">class:FastQC</a></td>
<td>FastQC and associated classes</td></tr>
<tr><td width="25%"><a href="regionStats.html">class:RegionStats</a></td>
<td>Find Regions of significance in microarray data</td></tr>
<tr><td width="25%"><a href="FastQC-class.html">class:SequenceQC</a></td>
<td>FastQC and associated classes</td></tr>
<tr><td width="25%"><a href="FastQC-class.html">class:SequenceQCSet</a></td>
<td>FastQC and associated classes</td></tr>
<tr><td width="25%"><a href="ClusteredCoverageList-class.html">ClusteredCoverageList</a></td>
<td>Container for coverage matrices with clustering results.</td></tr>
<tr><td width="25%"><a href="ClusteredCoverageList-class.html">ClusteredCoverageList-class</a></td>
<td>Container for coverage matrices with clustering results.</td></tr>
<tr><td width="25%"><a href="ClusteredCoverageList-class.html">ClusteredCoverageList-method</a></td>
<td>Container for coverage matrices with clustering results.</td></tr>
<tr><td width="25%"><a href="clusterPlots.html">clusterPlots</a></td>
<td>Visualisation of tables of feature coverages.</td></tr>
<tr><td width="25%"><a href="clusterPlots.html">clusterPlots-method</a></td>
<td>Visualisation of tables of feature coverages.</td></tr>
<tr><td width="25%"><a href="cpgBoxPlots.html">cpgBoxplots</a></td>
<td>Boxplots of intensity, binned by Cpg Density</td></tr>
<tr><td width="25%"><a href="cpgBoxPlots.html">cpgBoxplots-method</a></td>
<td>Boxplots of intensity, binned by Cpg Density</td></tr>
<tr><td width="25%"><a href="cpgDensityCalc.html">cpgDensityCalc</a></td>
<td>Calculate CpG Density in a Window</td></tr>
<tr><td width="25%"><a href="cpgDensityCalc.html">cpgDensityCalc-method</a></td>
<td>Calculate CpG Density in a Window</td></tr>
<tr><td width="25%"><a href="cpgDensityPlot.html">cpgDensityPlot</a></td>
<td>Plot the distribution of sequencing reads CpG densities.</td></tr>
<tr><td width="25%"><a href="cpgDensityPlot.html">cpgDensityPlot-method</a></td>
<td>Plot the distribution of sequencing reads CpG densities.</td></tr>
<tr><td width="25%"><a href="ChromaResults-class.html">cutoff</a></td>
<td>ChromaResults class</td></tr>
<tr><td width="25%"><a href="ChromaResults-class.html">cutoff-method</a></td>
<td>ChromaResults class</td></tr>
</table>

<h2><a name="D">-- D --</a></h2>

<table width="100%">
<tr><td width="25%"><a href="doSeqStats.html">doSeqStats</a></td>
<td>Calculate Statistics for Sequencing Data</td></tr>
<tr><td width="25%"><a href="doSeqStats.html">doSeqStats-method</a></td>
<td>Calculate Statistics for Sequencing Data</td></tr>
</table>

<h2><a name="E">-- E --</a></h2>

<table width="100%">
<tr><td width="25%"><a href="enrichmentCalc.html">enrichmentCalc</a></td>
<td>Calculate sequencing enrichment</td></tr>
<tr><td width="25%"><a href="enrichmentCalc.html">enrichmentCalc-method</a></td>
<td>Calculate sequencing enrichment</td></tr>
<tr><td width="25%"><a href="enrichmentPlot.html">enrichmentPlot</a></td>
<td>Plot the distribution of sequencing enrichment.</td></tr>
<tr><td width="25%"><a href="enrichmentPlot.html">enrichmentPlot-method</a></td>
<td>Plot the distribution of sequencing enrichment.</td></tr>
<tr><td width="25%"><a href="expr.html">expr</a></td>
<td>Vector of expression differences</td></tr>
</table>

<h2><a name="F">-- F --</a></h2>

<table width="100%">
<tr><td width="25%"><a href="FastQC-class.html">FastQC-class</a></td>
<td>FastQC and associated classes</td></tr>
<tr><td width="25%"><a href="ChromaResults-class.html">FDRTable</a></td>
<td>ChromaResults class</td></tr>
<tr><td width="25%"><a href="ChromaResults-class.html">FDRTable-method</a></td>
<td>ChromaResults class</td></tr>
<tr><td width="25%"><a href="featureBlocks.html">featureBlocks</a></td>
<td>Make windows for distances around a reference point.</td></tr>
<tr><td width="25%"><a href="featureBlocks.html">featureBlocks-method</a></td>
<td>Make windows for distances around a reference point.</td></tr>
<tr><td width="25%"><a href="featureScores.html">featureScores</a></td>
<td>Get scores at regular sample points around genomic features.</td></tr>
<tr><td width="25%"><a href="featureScores.html">featureScores-method</a></td>
<td>Get scores at regular sample points around genomic features.</td></tr>
<tr><td width="25%"><a href="findClusters.html">findClusters</a></td>
<td>Find Clusters Epigenetically Modified Genes</td></tr>
</table>

<h2><a name="G">-- G --</a></h2>

<table width="100%">
<tr><td width="25%"><a href="gcContentCalc.html">gcContentCalc</a></td>
<td>Calculate The gcContent of a Region</td></tr>
<tr><td width="25%"><a href="gcContentCalc.html">gcContentCalc-method</a></td>
<td>Calculate The gcContent of a Region</td></tr>
<tr><td width="25%"><a href="GDL2GRL.html">GDL2GRL</a></td>
<td>Utility function to covert a GenomeDataList object into GRangesList objects.</td></tr>
<tr><td width="25%"><a href="GDL2GRL.html">GDL2GRL-method</a></td>
<td>Utility function to covert a GenomeDataList object into GRangesList objects.</td></tr>
<tr><td width="25%"><a href="genomeBlocks.html">genomeBlocks</a></td>
<td>Creates bins across a genome.</td></tr>
<tr><td width="25%"><a href="genomeBlocks.html">genomeBlocks-method</a></td>
<td>Creates bins across a genome.</td></tr>
<tr><td width="25%"><a href="genQC.html">genQC</a></td>
<td>Plot Quality Checking Information for Sequencing Data</td></tr>
<tr><td width="25%"><a href="genQC.html">genQC-method</a></td>
<td>Plot Quality Checking Information for Sequencing Data</td></tr>
<tr><td width="25%"><a href="getCN.html">getCN</a></td>
<td>Calculate Copy Number and Map To Enriched Regions</td></tr>
<tr><td width="25%"><a href="getCN.html">getCN-method</a></td>
<td>Calculate Copy Number and Map To Enriched Regions</td></tr>
<tr><td width="25%"><a href="getProbePositionsDf.html">getProbePositionsDf</a></td>
<td>Translate Affymetrix probe information in a table.</td></tr>
<tr><td width="25%"><a href="getProbePositionsDf.html">getProbePositionsDf-method</a></td>
<td>Translate Affymetrix probe information in a table.</td></tr>
</table>

<h2><a name="L">-- L --</a></h2>

<table width="100%">
<tr><td width="25%"><a href="ScoresList-class.html">length-method</a></td>
<td>Container for 'featureScores()' output.</td></tr>
<tr><td width="25%"><a href="loadPairFile.html">loadPairFile</a></td>
<td>A routine to read Nimblegen tiling array intensities</td></tr>
<tr><td width="25%"><a href="loadSampleDirectory.html">loadSampleDirectory</a></td>
<td>A routine to read Nimblegen tiling array intensities</td></tr>
</table>

<h2><a name="M">-- M --</a></h2>

<table width="100%">
<tr><td width="25%"><a href="makeWindowLookupTable.html">makeWindowLookupTable</a></td>
<td>Using the output of 'annotationLookup', create a tabular storage of the indices</td></tr>
<tr><td width="25%"><a href="mappabilityCalc.html">mappabilityCalc</a></td>
<td>Calculate The Mappability of a Region</td></tr>
<tr><td width="25%"><a href="mappabilityCalc.html">mappabilityCalc-method</a></td>
<td>Calculate The Mappability of a Region</td></tr>
<tr><td width="25%"><a href="mergeReplicates.html">mergeReplicates</a></td>
<td>Merge GRanges that are of replicate experiments.</td></tr>
<tr><td width="25%"><a href="mergeReplicates.html">mergeReplicates-method</a></td>
<td>Merge GRanges that are of replicate experiments.</td></tr>
<tr><td width="25%"><a href="FastQC-class.html">Mismatches</a></td>
<td>FastQC and associated classes</td></tr>
<tr><td width="25%"><a href="FastQC-class.html">Mismatches-method</a></td>
<td>FastQC and associated classes</td></tr>
<tr><td width="25%"><a href="FastQC-class.html">MismatchTable</a></td>
<td>FastQC and associated classes</td></tr>
<tr><td width="25%"><a href="FastQC-class.html">MismatchTable-method</a></td>
<td>FastQC and associated classes</td></tr>
<tr><td width="25%"><a href="multiHeatmap.html">multiHeatmap</a></td>
<td>Superfigure plots</td></tr>
</table>

<h2><a name="N">-- N --</a></h2>

<table width="100%">
<tr><td width="25%"><a href="ScoresList-class.html">names-method</a></td>
<td>Container for 'featureScores()' output.</td></tr>
<tr><td width="25%"><a href="ScoresList-class.html">names&lt;--method</a></td>
<td>Container for 'featureScores()' output.</td></tr>
</table>

<h2><a name="O">-- O --</a></h2>

<table width="100%">
<tr><td width="25%"><a href="FastQC-class.html">Overrepresented_sequences</a></td>
<td>FastQC and associated classes</td></tr>
<tr><td width="25%"><a href="FastQC-class.html">Overrepresented_sequences-method</a></td>
<td>FastQC and associated classes</td></tr>
</table>

<h2><a name="P">-- P --</a></h2>

<table width="100%">
<tr><td width="25%"><a href="FastQC-class.html">Per_base_GC_content</a></td>
<td>FastQC and associated classes</td></tr>
<tr><td width="25%"><a href="FastQC-class.html">Per_base_GC_content-method</a></td>
<td>FastQC and associated classes</td></tr>
<tr><td width="25%"><a href="FastQC-class.html">Per_base_N_content</a></td>
<td>FastQC and associated classes</td></tr>
<tr><td width="25%"><a href="FastQC-class.html">Per_base_N_content-method</a></td>
<td>FastQC and associated classes</td></tr>
<tr><td width="25%"><a href="FastQC-class.html">Per_base_sequence_content</a></td>
<td>FastQC and associated classes</td></tr>
<tr><td width="25%"><a href="FastQC-class.html">Per_base_sequence_content-method</a></td>
<td>FastQC and associated classes</td></tr>
<tr><td width="25%"><a href="FastQC-class.html">Per_base_sequence_quality</a></td>
<td>FastQC and associated classes</td></tr>
<tr><td width="25%"><a href="FastQC-class.html">Per_base_sequence_quality-method</a></td>
<td>FastQC and associated classes</td></tr>
<tr><td width="25%"><a href="FastQC-class.html">Per_sequence_GC_content</a></td>
<td>FastQC and associated classes</td></tr>
<tr><td width="25%"><a href="FastQC-class.html">Per_sequence_GC_content-method</a></td>
<td>FastQC and associated classes</td></tr>
<tr><td width="25%"><a href="FastQC-class.html">Per_sequence_quality_scores</a></td>
<td>FastQC and associated classes</td></tr>
<tr><td width="25%"><a href="FastQC-class.html">Per_sequence_quality_scores-method</a></td>
<td>FastQC and associated classes</td></tr>
<tr><td width="25%"><a href="plotClusters.html">plotClusters</a></td>
<td>Plot Scores of Cluster Regions</td></tr>
<tr><td width="25%"><a href="plotClusters.html">plotClusters-method</a></td>
<td>Plot Scores of Cluster Regions</td></tr>
<tr><td width="25%"><a href="processNDF.html">processNDF</a></td>
<td>Reads in a Nimblegen microarray design file (NDF)</td></tr>
</table>

<h2><a name="R">-- R --</a></h2>

<table width="100%">
<tr><td width="25%"><a href="FastQC-class.html">readFastQC</a></td>
<td>FastQC and associated classes</td></tr>
<tr><td width="25%"><a href="ChromaResults-class.html">regions</a></td>
<td>ChromaResults class</td></tr>
<tr><td width="25%"><a href="ChromaResults-class.html">regions-method</a></td>
<td>ChromaResults class</td></tr>
<tr><td width="25%"><a href="regionStats.html">regionStats</a></td>
<td>Find Regions of significance in microarray data</td></tr>
<tr><td width="25%"><a href="regionStats.html">RegionStats-class</a></td>
<td>Find Regions of significance in microarray data</td></tr>
<tr><td width="25%"><a href="regionStats.html">regionStats-method</a></td>
<td>Find Regions of significance in microarray data</td></tr>
</table>

<h2><a name="S">-- S --</a></h2>

<table width="100%">
<tr><td width="25%"><a href="samplesList.html">samplesList</a></td>
<td>Short Reads from Cancer and Normal</td></tr>
<tr><td width="25%"><a href="ScoresList-class.html">ScoresList</a></td>
<td>Container for 'featureScores()' output.</td></tr>
<tr><td width="25%"><a href="ScoresList-class.html">ScoresList-class</a></td>
<td>Container for 'featureScores()' output.</td></tr>
<tr><td width="25%"><a href="sequenceCalc.html">sequenceCalc</a></td>
<td>Find occurences of a DNA pattern</td></tr>
<tr><td width="25%"><a href="sequenceCalc.html">sequenceCalc-method</a></td>
<td>Find occurences of a DNA pattern</td></tr>
<tr><td width="25%"><a href="FastQC-class.html">SequenceQC-class</a></td>
<td>FastQC and associated classes</td></tr>
<tr><td width="25%"><a href="FastQC-class.html">SequenceQCSet</a></td>
<td>FastQC and associated classes</td></tr>
<tr><td width="25%"><a href="FastQC-class.html">SequenceQCSet-class</a></td>
<td>FastQC and associated classes</td></tr>
<tr><td width="25%"><a href="FastQC-class.html">Sequence_Duplication_Levels</a></td>
<td>FastQC and associated classes</td></tr>
<tr><td width="25%"><a href="FastQC-class.html">Sequence_Duplication_Levels-method</a></td>
<td>FastQC and associated classes</td></tr>
<tr><td width="25%"><a href="FastQC-class.html">Sequence_Length_Distribution</a></td>
<td>FastQC and associated classes</td></tr>
<tr><td width="25%"><a href="FastQC-class.html">Sequence_Length_Distribution-method</a></td>
<td>FastQC and associated classes</td></tr>
<tr><td width="25%"><a href="ChromaResults-class.html">show-method</a></td>
<td>ChromaResults class</td></tr>
<tr><td width="25%"><a href="ClusteredCoverageList-class.html">show-method</a></td>
<td>Container for coverage matrices with clustering results.</td></tr>
<tr><td width="25%"><a href="FastQC-class.html">show-method</a></td>
<td>FastQC and associated classes</td></tr>
<tr><td width="25%"><a href="ScoresList-class.html">show-method</a></td>
<td>Container for 'featureScores()' output.</td></tr>
<tr><td width="25%"><a href="regionStats.html">show-method</a></td>
<td>Find Regions of significance in microarray data</td></tr>
<tr><td width="25%"><a href="significancePlots.html">significancePlots</a></td>
<td>Create line plots of averaged signal across a promoter compared to random sampling</td></tr>
<tr><td width="25%"><a href="significancePlots.html">significancePlots-method</a></td>
<td>Create line plots of averaged signal across a promoter compared to random sampling</td></tr>
</table>

<h2><a name="T">-- T --</a></h2>

<table width="100%">
<tr><td width="25%"><a href="ScoresList-class.html">tables</a></td>
<td>Container for 'featureScores()' output.</td></tr>
<tr><td width="25%"><a href="ScoresList-class.html">tables-method</a></td>
<td>Container for 'featureScores()' output.</td></tr>
</table>

<h2><a name="W">-- W --</a></h2>

<table width="100%">
<tr><td width="25%"><a href="writeWig.html">writeWig</a></td>
<td>Writes sequencing data out into a wiggle files</td></tr>
<tr><td width="25%"><a href="writeWig.html">writeWig-method</a></td>
<td>Writes sequencing data out into a wiggle files</td></tr>
</table>

<h2><a name="misc">-- misc --</a></h2>

<table width="100%">
<tr><td width="25%"><a href="ClusteredCoverageList-class.html">[-method</a></td>
<td>Container for coverage matrices with clustering results.</td></tr>
<tr><td width="25%"><a href="FastQC-class.html">[-method</a></td>
<td>FastQC and associated classes</td></tr>
<tr><td width="25%"><a href="ScoresList-class.html">[-method</a></td>
<td>Container for 'featureScores()' output.</td></tr>
</table>
</body></html>
